### DISCUSSION TITLE
How to differentiate Internal and External Team sites using PnP Search Results Webpart

### DISCUSSION THREAD

**Discussion by monadha** [microsoft-search/pnp-modern-search#1370] [Q&A] [answered]
Hi,
I am currently using the PnP Search Results webpart to display all the Team sites using custom KQL Query.

Now I want to differentiate the Search Results based on Site Sharing Capability and display Internal/External Team sites separately on the search result webpart.

I have tried using  "**ViewableByExternalUsers**"  property is the KQL Query template. When ViewableByExternalUsers:false, It should display "**Internal sites**"  and when ViewableByExternalUsers:true, it should display "**External sites**" BUT it does not display the correct search results. 

I am not sure if I am using the correct property to distinguish between Internal/External sites or am I missing something? 
Has anyone have worked on this issue or is aware of how to differentiate between Internal/External Team sites in PnP Search Results. Let me know.

Thanks.

**Comment by jhholm** [comment]
Cannot give a straight answer, but I would begin by using an external open source tool such as [SP Editor](https://chrome.google.com/webstore/detail/sp-editor/ecblfcmjnbbgaojblcpmjoamegpbodhd?hl=en) or [SharePoint Search Query Tool](https://github.com/pnp/PnP-Tools/tree/master/Solutions/SharePoint.Search.QueryTool). Both enable you to generate search queries and fetch most of the provided managed properties for a search result.

I would check the managed properties for those two use cases and see the differences that could give a semantical answer. If you can see that **ViewableByExternalUsers** is actually a property that shows the information, check if the returned value is actually a boolean and thus your KQL is correct. You should also check if the managed property is queryable through SharePoint Search Admin. If this is not the case, you should be able to create your own Managed Property that is mapped to the same Crawled Property that is behind the out of the box Managed Property.

**Comment by monadha** [comment]
Hi Janne,

ViewableByExternalUsers returns boolean value true/false.  Few sites show the correct results but few sites it doesn't show the correct results. I have followed this article- https://docs.microsoft.com/en-us/microsoft-365/compliance/keyword-queries-and-search-conditions?view=o365-worldwide and tested the ViewableByExternalUsers property but it doesn't give an accurate result even after re-indexing the site.

Is there anything else I should be checking to filter the Internal/External SharePoint site?

**Comment by wobba** [comment, marked as answer]
Answered in #1355
